trl/trainer/ppov2_bandit_rloo_trainer.py

```diff
@@ -0,0 +1,704 @@
+        self.data_collator = data_collator
+        self.eval_dataset = eval_dataset
+        self.eval_generation_config = eval_generation_config
+        if eval_generation_config is None:
+            self.eval_generation_config = train_generation_config
+        self.optimizer, self.lr_scheduler = optimizers
+        self.callbacks = callbacks
+
+        #########
+        # calculate various batch sizes
+        #########
+        accelerator = Accelerator(gradient_accumulation_steps=args.gradient_accumulation_steps)
+        self.accelerator = accelerator
+        args.world_size = accelerator.num_processes
+        args.local_batch_size = args.per_device_train_batch_size * args.gradient_accumulation_steps * args.nminibatches
+        args.micro_batch_size = int(args.per_device_train_batch_size * args.world_size)
+        args.batch_size = int(args.local_batch_size * args.world_size)
+        args.mini_batch_size = exact_div(args.batch_size, args.nminibatches)
+        args.local_mini_batch_size = exact_div(args.local_batch_size, args.nminibatches)
+        if args.whiten_rewards:
+            assert (
+                args.local_mini_batch_size >= 8
+            ), f"Per-rank minibatch size {args.local_mini_batch_size} is insufficient for whitening"
+        # `per_rank_rollout_batch_size` is our `args.local_batch_size`
+        # `per_rank_minibatch_size` is our `args.local_mini_batch_size`
+        args.num_updates = args.total_episodes // args.batch_size
+        time_tensor = torch.tensor(int(time.time()), device=accelerator.device)
+        time_int = broadcast(time_tensor, 0).item()  # avoid different timestamps across processes
+        args.run_name = f"{args.exp_name}__{args.seed}__{time_int}"
+        self.local_seed = args.seed + accelerator.process_index * 100003  # Prime
+
+        #########
+        # disable dropout
+        #########
+        for module in [policy, ref_policy, reward_model]:
+            disable_dropout(module)
+        policy.generation_config.eos_token_id = (
+            None  # disable `pad_token_id` and `eos_token_id` because we just want to
+        )
+        policy.generation_config.pad_token_id = None  # generate tokens without truncation / padding
+
+        #########
+        # setup model, optimizer, and others
+        #########
+        if args.truncate_token and args.truncate_token == "eos":
+            args.truncate_token_id = tokenizer.eos_token_id
+        self.model = policy
+        self.create_optimizer_and_scheduler(num_training_steps=args.num_updates)
+
+        #########
+        ### trainer specifics
+        #########
+        self.state = TrainerState(
+            is_local_process_zero=self.is_local_process_zero(),
+            is_world_process_zero=self.is_world_process_zero(),
+        )
+        DEFAULT_CALLBACKS = [DefaultFlowCallback]
+        default_callbacks = DEFAULT_CALLBACKS + get_reporting_integration_callbacks(self.args.report_to)
```

Comment:
Can we move most of this to just being a `super().__init__()` call? This will prevent the need for maintenance which is already handled by the base `Trainer` class.

Reply:
`super().__init__()` is exactly what I tried, but I remember running into some arcane errors. Will try it out again. TODO.